### PR TITLE
Virus code is getting gutted S O O N

### PR DESCRIPTION
**Pull request by Iamgoofball**
oh my god this is awful

### CODE REVIEW

**Comment by DoctoreMad** [comment]
>oh my god this is awful

Is it worse than you though

**Comment by banthebantz** [comment]
the point is that virus code is so bad it is a fucking virus.

File: code/datums/diseases/advance/advance.dm [hunk 1]
```diff
@@ -385,7 +400,7 @@
 				if(!D.HasSymptom(S))
 					D.symptoms += S
 					i -= 1
-	while(i > 0)
```
Comment (Razharas) [inline code/datums/diseases/advance/advance.dm]: You are making it worse optimization-wise
Id say you can make it faster just checking it to not be 0 but its byond so i dunno how it does it and its a asm level of microoptimization

Comment (Cyberboss) [inline code/datums/diseases/advance/advance.dm]: Leave it like this, it's not possible to have a disease without a symptom

Comment (RemieRichards) [inline code/datums/diseases/advance/advance.dm]: The do-while is best here.
@Cyberboss Not relevant, the input doesn't let you make symptom-less viruses anyway.

Comment (Iamgoofball) [inline code/datums/diseases/advance/advance.dm]: This actually works faster? What the fuck

File: code/datums/diseases/advance/advance.dm [hunk 2]
```diff
@@ -57,16 +57,18 @@
 		if(!D || !D.symptoms || !D.symptoms.len)
 			symptoms = GenerateSymptoms(0, 2)
 		else
-			for(var/datum/symptom/S in D.symptoms)
-				symptoms += new S.type
+			for(var/S in D.symptoms)
+				var/datum/symptom/SY
```
Comment (lzimann) [inline code/datums/diseases/advance/advance.dm]: This will runtime. Needs to assign S to SY

File: code/datums/diseases/advance/advance.dm (hunk 1 above)
Comment (RemieRichards) [inline code/datums/diseases/advance/advance.dm]: @Iamgoofball of course it does? the ```do``` means the block of code always runs once (which the define is non-zero so it should always run atleast ONCE), moving the while to the top would mean the condition (```i > 0```) would be checked ```SYMPTOM_LIMIT``` times, where as with the do the condition is only checked ```SYMPTOM_LIMIT-1``` times.

File: code/datums/diseases/advance/advance.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/datums/diseases/advance/advance.dm]: whooops

File: code/datums/diseases/advance/advance.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/datums/diseases/advance/advance.dm]: b y o n d

Comment (ohnopigeons) [inline code/datums/diseases/advance/advance.dm]: This has NOTHING to do with byond, it's literally do-while vs while

**Comment by lzimann** [comment]
goof pls

**Comment by ohnopigeons** [comment]
```
for(var/SY in possible_symptoms)
	var/datum/symptom/S = SY
```

What does this change do exactly?

**Comment by AnturK** [comment]
Typed loops are marginally slower, so if you have pre knowledge that list only contains one type , typeless loop + cast is faster.

**Comment by lzimann** [comment]
Just ping whenever you want this reopen.

**Pull request closed without merging**